### PR TITLE
chore: fix JSDoc comments in vehicle.ts

### PR DESCRIPTION
**Pull request by pkuczynski**
Refs #203

### CODE REVIEW

File: src/vehicle.ts [hunk 1]
```diff
@@ -14,72 +14,17 @@ export class Vehicle {
...
-        'Recumbent Bicycle',
-      ],
-    };
   }
 
   /**
-   * vehicle
-   *
-   * @method faker.vehicle.vehicle
+   * Returns a random vehicle.
```
Comment (prisis) [inline src/vehicle.ts:20]: Can we use [https://jsdoc.app/tags-example.html](https://jsdoc.app/tags-example.html) and move the sampleResults to it?

Comment (pkuczynski) [reply src/vehicle.ts:20]: I was wondering if that makes sense? I can update the PR. Let me know if I should?

Comment (prisis) [reply src/vehicle.ts:20]: I think it's easier to see what it will return :) for new developer, lets wait what the other maintainer will say

Comment (Shinigami92) [reply src/vehicle.ts:20]: This is a really good idea 👍

**Comment by Shinigami92** [comment]
Suggested edit:
```diff
diff --git a/src/vehicle.ts b/src/vehicle.ts
--- src/vehicle.ts
+++ src/vehicle.ts
@@ -13,112 +13,61 @@
         continue;
       }
       this[name] = this[name].bind(this);
     }
-
-    // TODO @Shinigami92 2022-01-13: Find better strategy
-    // @ts-expect-error
-    this.vehicle.schema = {
-      description: 'Generates a random vehicle.',
-      sampleResults: ['BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'],
-    };
-    // @ts-expect-error
-    this.manufacturer.schema = {
-      description: 'Generates a manufacturer name.',
-      sampleResults: ['Ford', 'Jeep', 'Tesla'],
-    };
-    // @ts-expect-error
-    this.model.schema = {
-      description: 'Generates a vehicle model.',
-      sampleResults: ['Explorer', 'Camry', 'Ranchero'],
-    };
-    // @ts-expect-error
-    this.type.schema = {
-      description: 'Generates a vehicle type.',
-      sampleResults: ['Coupe', 'Convertable', 'Sedan', 'SUV'],
-    };
-    // @ts-expect-error
-    this.fuel.schema = {
-      description: 'Generates a fuel type.',
-      sampleResults: ['Electric', 'Gasoline', 'Diesel'],
-    };
-    // @ts-expect-error
-    this.vin.schema = {
-      description: 'Generates a valid VIN number.',
-      sampleResults: ['YV1MH682762184654', '3C7WRMBJ2EG208836'],
-    };
-    // @ts-expect-error
-    this.color.schema = {
-      description: 'Generates a color',
-      sampleResults: ['red', 'white', 'black'],
-    };
-    // @ts-expect-error
-    this.vrm.schema = {
-      description: 'Generates a vehicle vrm',
-      sampleResults: ['MF56UPA', 'GL19AAQ', 'SF20TTA'],
-    };
-    // @ts-expect-error
-    this.bicycle.schema = {
-      description: 'Generates a type of bicycle',
-      sampleResults: [
-        'Adventure Road Bicycle',
-        'City Bicycle',
-        'Recumbent Bicycle',
-      ],
-    };
   }
 
   /**
-   * vehicle
+   * Returns a random vehicle.
    *
-   * @method faker.vehicle.vehicle
+   * @example 'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
    */
   vehicle(): string {
     return fake('{{vehicle.manufacturer}} {{vehicle.model}}');
   }
 
   /**
-   * manufacturer
+   * Returns a manufacturer name.
    *
-   * @method faker.vehicle.manufacturer
+   * @example 'Ford', 'Jeep', 'Tesla'
    */
   manufacturer(): string {
     return this.faker.random.arrayElement(
       this.faker.definitions.vehicle.manufacturer
     );
   }
 
   /**
-   * model
+   * Returns a vehicle model.
    *
-   * @method faker.vehicle.model
+   * @example 'Explorer', 'Camry', 'Ranchero'
    */
   model(): string {
     return this.faker.random.arrayElement(this.faker.definitions.vehicle.model);
   }
 
   /**
-   * type
+   * Returns a vehicle type.
    *
-   * @method faker.vehicle.type
+   * @example 'Coupe', 'Convertable', 'Sedan', 'SUV'
    */
   type(): string {
     return this.faker.random.arrayElement(this.faker.definitions.vehicle.type);
   }
 
   /**
-   * fuel
+   * Returns a fuel type.
    *
-   * @method faker.vehicle.fuel
+   * @example 'Electric', 'Gasoline', 'Diesel'
    */
   fuel(): string {
     return this.faker.random.arrayElement(this.faker.definitions.vehicle.fuel);
   }
 
   /**
-   * vin
+   * Returns a valid VIN number.
    *
-   * @method faker.vehicle.vin
+   * @example 'YV1MH682762184654', '3C7WRMBJ2EG208836'
    */
   vin(): string {
     const bannedChars = ['o', 'i', 'q'];
     return (
@@ -134,20 +83,20 @@
       .toUpperCase();
   }
 
   /**
-   * color
+   * Returns a vehicle color.
    *
-   * @method faker.vehicle.color
+   * @example 'red', 'white', 'black'
    */
   color(): string {
     return fake('{{commerce.color}}');
   }
 
   /**
-   * vrm
+   * Returns a vehicle vrm.
    *
-   * @method faker.vehicle.vrm
+   * @example 'MF56UPA', 'GL19AAQ', 'SF20TTA'
    */
   vrm(): string {
     return (
       this.faker.random.alpha({ count: 2, upcase: true }) +
@@ -157,11 +106,11 @@
     ).toUpperCase();
   }
 
   /**
-   * bicycle
+   * Returns a type of bicycle.
    *
-   * @method faker.vehicle.bicycle
+   * @example 'Adventure Road Bicycle', 'City Bicycle', 'Recumbent Bicycle'
    */
   bicycle(): string {
     return this.faker.random.arrayElement(
       this.faker.definitions.vehicle.bicycle_type

```

**Comment by Shinigami92** [comment]
Suggested edit:
```diff
diff --git a/src/vehicle.ts b/src/vehicle.ts
--- src/vehicle.ts
+++ src/vehicle.ts
@@ -13,112 +13,61 @@
         continue;
       }
       this[name] = this[name].bind(this);
     }
-
-    // TODO @Shinigami92 2022-01-13: Find better strategy
-    // @ts-expect-error
-    this.vehicle.schema = {
-      description: 'Generates a random vehicle.',
-      sampleResults: ['BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'],
-    };
-    // @ts-expect-error
-    this.manufacturer.schema = {
-      description: 'Generates a manufacturer name.',
-      sampleResults: ['Ford', 'Jeep', 'Tesla'],
-    };
-    // @ts-expect-error
-    this.model.schema = {
-      description: 'Generates a vehicle model.',
-      sampleResults: ['Explorer', 'Camry', 'Ranchero'],
-    };
-    // @ts-expect-error
-    this.type.schema = {
-      description: 'Generates a vehicle type.',
-      sampleResults: ['Coupe', 'Convertable', 'Sedan', 'SUV'],
-    };
-    // @ts-expect-error
-    this.fuel.schema = {
-      description: 'Generates a fuel type.',
-      sampleResults: ['Electric', 'Gasoline', 'Diesel'],
-    };
-    // @ts-expect-error
-    this.vin.schema = {
-      description: 'Generates a valid VIN number.',
-      sampleResults: ['YV1MH682762184654', '3C7WRMBJ2EG208836'],
-    };
-    // @ts-expect-error
-    this.color.schema = {
-      description: 'Generates a color',
-      sampleResults: ['red', 'white', 'black'],
-    };
-    // @ts-expect-error
-    this.vrm.schema = {
-      description: 'Generates a vehicle vrm',
-      sampleResults: ['MF56UPA', 'GL19AAQ', 'SF20TTA'],
-    };
-    // @ts-expect-error
-    this.bicycle.schema = {
-      description: 'Generates a type of bicycle',
-      sampleResults: [
-        'Adventure Road Bicycle',
-        'City Bicycle',
-        'Recumbent Bicycle',
-      ],
-    };
   }
 
   /**
-   * vehicle
+   * Returns a random vehicle.
    *
-   * @method faker.vehicle.vehicle
+   * @example 'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
    */
   vehicle(): string {
     return fake('{{vehicle.manufacturer}} {{vehicle.model}}');
   }
 
   /**
-   * manufacturer
+   * Returns a manufacturer name.
    *
-   * @method faker.vehicle.manufacturer
+   * @example 'Ford', 'Jeep', 'Tesla'
    */
   manufacturer(): string {
     return this.faker.random.arrayElement(
       this.faker.definitions.vehicle.manufacturer
     );
   }
 
   /**
-   * model
+   * Returns a vehicle model.
    *
-   * @method faker.vehicle.model
+   * @example 'Explorer', 'Camry', 'Ranchero'
    */
   model(): string {
     return this.faker.random.arrayElement(this.faker.definitions.vehicle.model);
   }
 
   /**
-   * type
+   * Returns a vehicle type.
    *
-   * @method faker.vehicle.type
+   * @example 'Coupe', 'Convertable', 'Sedan', 'SUV'
    */
   type(): string {
     return this.faker.random.arrayElement(this.faker.definitions.vehicle.type);
   }
 
   /**
-   * fuel
+   * Returns a fuel type.
    *
-   * @method faker.vehicle.fuel
+   * @example 'Electric', 'Gasoline', 'Diesel'
    */
   fuel(): string {
     return this.faker.random.arrayElement(this.faker.definitions.vehicle.fuel);
   }
 
   /**
-   * vin
+   * Returns a valid VIN number.
    *
-   * @method faker.vehicle.vin
+   * @example 'YV1MH682762184654', '3C7WRMBJ2EG208836'
    */
   vin(): string {
     const bannedChars = ['o', 'i', 'q'];
     return (
@@ -134,20 +83,20 @@
       .toUpperCase();
   }
 
   /**
-   * color
+   * Returns a vehicle color.
    *
-   * @method faker.vehicle.color
+   * @example 'red', 'white', 'black'
    */
   color(): string {
     return fake('{{commerce.color}}');
   }
 
   /**
-   * vrm
+   * Returns a vehicle vrm.
    *
-   * @method faker.vehicle.vrm
+   * @example 'MF56UPA', 'GL19AAQ', 'SF20TTA'
    */
   vrm(): string {
     return (
       this.faker.random.alpha({ count: 2, upcase: true }) +
@@ -157,11 +106,11 @@
     ).toUpperCase();
   }
 
   /**
-   * bicycle
+   * Returns a type of bicycle.
    *
-   * @method faker.vehicle.bicycle
+   * @example 'Adventure Road Bicycle', 'City Bicycle', 'Recumbent Bicycle'
    */
   bicycle(): string {
     return this.faker.random.arrayElement(
       this.faker.definitions.vehicle.bicycle_type

```

File: src/vehicle.ts [hunk 2]
```diff
@@ -14,72 +14,25 @@ export class Vehicle {
...
-        'City Bicycle',
-        'Recumbent Bicycle',
-      ],
-    };
   }
 
   /**
-   * vehicle
+   * Returns a random vehicle.
+   * @returns {string}
```
Comment (Shinigami92) [inline src/vehicle.ts:21]: Please remove all `@returns` due to TS already provide this out of the box

File: src/vehicle.ts [hunk 3]
```diff
@@ -14,72 +14,25 @@ export class Vehicle {
...
    *
-   * @method faker.vehicle.vehicle
+   * @example
+   * 'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
```
Comment (Shinigami92) [inline src/vehicle.ts:24]: Please reformat these to

```suggestion
   * @example 'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
```

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: src/vehicle.ts (hunk 3 above)
Comment (pkuczynski) [reply src/vehicle.ts]: Done

File: src/vehicle.ts (hunk 2 above)
Comment (pkuczynski) [reply src/vehicle.ts]: Done

**Review by Shinigami92 — APPROVED** [review]

File: src/vehicle.ts [hunk 4]
```diff
@@ -14,72 +14,21 @@ export class Vehicle {
...
-      ],
-    };
   }
 
   /**
-   * vehicle
+   * Returns a random vehicle.
    *
-   * @method faker.vehicle.vehicle
+   * @example 'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
```
Comment (prisis) [inline src/vehicle.ts:22]: Im relay to say it but you need to change the examples... see https://stackoverflow.com/questions/2961310/jsdoc-adding-real-code-in-documentation

the correct example would be

@example
// returns 'BMW Explorer'
faker.vehicle.vehicle()
// returns 'Ford Camry'
faker.vehicle.vehicle()

Comment (pkuczynski) [reply src/vehicle.ts:22]: I am not sure which tool are you planning to use to generate the docs, so I only tested in WebStorm IDE:

1. My original version

```
@example
'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
```

Screenshot 2022-01-24 at 19 35 45


2. After a suggestion from @Shinigami92 

```
@example 'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
```

Screenshot 2022-01-24 at 19 36 00

3. 

```
@example
faker.vehicle.vehicle() // 'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
```

Screenshot 2022-01-24 at 19 35 25


4.

```
@example
// returns 'BMW Explorer', 'Ford Camry', 'Lamborghini Ranchero'
faker.vehicle.vehicle()
```

Screenshot 2022-01-24 at 19 35 07

5. Version from @prisis 

```
@example
// returns 'BMW Explorer'
faker.vehicle.vehicle()

// returns 'Ford Camry'
faker.vehicle.vehicle()

// returns 'Lamborghini Ranchero'
faker.vehicle.vehicle()
```

Screenshot 2022-01-24 at 19 39 20

So guys, could you please decide on the final format and I will then follow this suggestion?

For me the word `returns` is a bit obsolete... But also having 3 almost identical examples seems to much...

Comment (Shinigami92) [reply src/vehicle.ts:22]: For the `@example` in VSCode it produce the same output, so use whats better for WebStorm :+1:

Comment (pkuczynski) [reply src/vehicle.ts:22]: For me 3. or 5. makes most sense but only with one example

Comment (prisis) [reply src/vehicle.ts:22]: I like 3 :D it looks good in phpstorm too

Comment (pkuczynski) [reply src/vehicle.ts:22]: One more version, which is my ultimate preference as it nicely colors strings

6. 

```
@example
faker.vehicle.vehicle()
'BMW Explorer'
```

Screenshot 2022-01-25 at 00 14 57

Comment (pkuczynski) [reply src/vehicle.ts]: I changed it to 3. to get it merged faster...

File: src/vehicle.ts [hunk 5]
```diff
@@ -88,36 +39,40 @@ export class Vehicle {
   }
 
   /**
-   * model
+   * Returns a vehicle model.
    *
-   * @method faker.vehicle.model
+   * @example 'Explorer', 'Camry', 'Ranchero'
```
Comment (prisis) [inline src/vehicle.ts:44]: Did you forget this line?

Comment (pkuczynski) [reply src/vehicle.ts]: good catch! thanks!

**Review by Shinigami92 — DISMISSED** [review]

File: src/vehicle.ts [hunk 6]
```diff
@@ -88,36 +39,40 @@ export class Vehicle {
...
+   * @example
+   * faker.vehicle.fuel() // 'Electric'
    */
   fuel(): string {
     return this.faker.random.arrayElement(this.faker.definitions.vehicle.fuel);
   }
 
   /**
-   * vin
+   * Returns a valid VIN number.
```
Comment (ST-DDT) [inline src/vehicle.ts:72]: Maybe explain what a `VIN` is. Maybe like this `Vehicle Identification Number (VIN)`?

File: src/vehicle.ts [hunk 7]
```diff
@@ -135,18 +90,20 @@ export class Vehicle {
...
+   * @example
+   * faker.vehicle.color() // 'red'
    */
   color(): string {
     return fake('{{commerce.color}}');
   }
 
   /**
-   * vrm
+   * Returns a vehicle vrm.
```
Comment (ST-DDT) [inline src/vehicle.ts:103]: Maybe explain what a VRM is. Is it a vehicle license plate?

**Review by ST-DDT — CHANGES_REQUESTED** [review]
Looks good to me. Just some minor nit picking for non native Americans.

File: src/vehicle.ts (hunk 6 above)
Comment (pkuczynski) [reply src/vehicle.ts:72]: VIN is international I think?. Added desc, please check again.

File: src/vehicle.ts (hunk 7 above)
Comment (pkuczynski) [reply src/vehicle.ts:103]: Yes! Vehicle Registration Mark. Desc added.

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**